subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -1231,7 +1231,7 @@ static int tls_mbedtls_handshake(struct tls_context *context,
 			/* MbedTLS API documentation requires session to
 			 * be reset in other error cases
 			 */
-			NET_ERR("TLS handshake error: -%x", -ret);
+			NET_ERR("TLS handshake error: 0x%x", -ret);
```

Comment:
+1 to adding 0x there, but I am not sure if removing the "-" is good. The mbed error values are <0 but when printing them we want to print as >0 (so the reason for `-ret`) and we add "-" to printout.
Example:
```
#include <stdio.h>

#define MBEDTLS_ERR_SSL_WANT_READ                         -0x6900

void main(void)
{
	int ret = MBEDTLS_ERR_SSL_WANT_READ;

	printf("ret = 0x%x\n", -ret);
	printf("ret = -0x%x\n", -ret);
}
```
would output
```
ret = 0x6900
ret = -0x6900
```
IMHO the latter is better because it will match the error definition fully.